pandas/tests/base/conftest.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+import pytest
+
+import pandas as pd
+from pandas.tests.indexes.conftest import indices_dict
+
+
+def _create_series(index):
+    """ Helper for the _series dict """
+    data = np.random.randn(len(index))
+    return pd.Series(data, index=index, name=index.name)
+
+
+_series = {
+    f"series-with-{i_id}-index": _create_series(i) for i_id, i in indices_dict.items()
+}
+
+
+def _create_narrow_series(data_dtype):
+    """ Helper for the _narrow_series dict """
+    index = indices_dict["int"].copy()
+    size = len(index)
+    if np.issubdtype(data_dtype, np.floating):
+        data = np.random.choice(size, size=size, replace=False)
+    elif np.issubdtype(data_dtype, np.integer):
+        data = np.random.randn(size)
+    else:
+        raise ValueError(f"Received an unexpected data_dtype: {data_dtype}")
+
+    return pd.Series(data.astype(data_dtype), index=index, name="a")
+
+
+_narrow_series = {
+    "float32-series": _create_narrow_series(np.float32),
+    "int8-series": _create_narrow_series(np.int8),
+    "int16-series": _create_narrow_series(np.int16),
+    "int32-series": _create_narrow_series(np.int32),
+    "uint8-series": _create_narrow_series(np.uint8),
+    "uint16-series": _create_narrow_series(np.uint16),
+    "uint32-series": _create_narrow_series(np.uint32),
+}
+
+_all_objs = {**indices_dict, **_series, **_narrow_series}
+
+
+@pytest.fixture(params=_all_objs.keys())
+def index_or_series_obj(request):
```

Comment:
Hmm I think this is a rather complicated way of composing things. Given there is already a fixture for `indeces` that does something very similar can we just not modify this to use that and create the Series objects as required?

Reply:
I'm not sure I understand you right. What do you mean by "create the Series objects as required"?

I need a fixture to replace [self.all_objs](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/base/test_ops.py#L83) from the `Ops` mixin. This will be used in a lot more places in `pandas/tests/base/test_ops.py`, so I can't just construct the Series' in the test cases.

I was hoping that there are other fixtures which I could use instead of `_series` and `_narrow_series`, but I think I still know too little about the codebase to find them. I'd be grateful for any hints to point me in the right direction ✌️

Reply:
To rephrase I'm asking if there is a way to leverage the already existing indices fixture, as this duplicates part of the logic of that

Reply:
Not really. I checked two possibilities:
1. Creating more parametrized fixtures for series and narrow_series, something like:
```
@pytest.fixture
def series(indices):
    data = np.random.randn(len(indices))
    return pd.Series(data, index=indices, name=indices.name)
```
When using these fixtures though, the product of all test cases is executed instead of the sequence. So a of redundant test cases are run.

2. Letting `index_or_series_obj` depend on `indices`:
The problem is that `index_or_series_obj` is supposed to have a different length. So this doesn't really work.

Based on this, the current implementation seems to be the best we can do.

Reply:
Issue 1 and a potential solution (requires a new dependency) is well explained in [this SO post](https://stackoverflow.com/questions/24340681/how-to-concatenate-several-parametrized-fixtures-into-a-new-fixture-in-py-test).